mountpoint-s3/src/inode/negative_cache.rs

```diff
@@ -0,0 +1,204 @@
+use std::time::{Duration, Instant};
+
+use linked_hash_map::LinkedHashMap;
+
+use super::{expiry::Expiry, InodeNo};
+
+use crate::sync::RwLock;
+
+/// A caches for negative lookups.
+/// Maintains a bounded set of (parent_ino, child_name) entries that expire after a fixed time.
+#[derive(Debug)]
+pub struct NegativeCache {
+    /// Holds keys in expiration order from oldest to newest.
+    map: RwLock<LinkedHashMap<Key, Expiry>>,
+    /// Upper bound for the cache.
+    max_size: usize,
+    /// TTL of a key at insertion.
+    ttl: Duration,
```

Comment:
Yes, that is the assumption for this structure. In fact, the implementation relies on it because it means that the insertion order is also the expiration order.